dht.go

```diff
@@ -133,33 +149,86 @@ func NewDHTClient(ctx context.Context, h host.Host, dstore ds.Batching) *IpfsDHT
 
 func makeDHT(ctx context.Context, h host.Host, dstore ds.Batching, protocols []protocol.ID) *IpfsDHT {
 	rt := kb.NewRoutingTable(KValue, kb.ConvertPeerID(h.ID()), time.Minute, h.Peerstore())
+	rtRecoveryChan := make(chan *rtRecoveryReq)
 
 	cmgr := h.ConnManager()
+
 	rt.PeerAdded = func(p peer.ID) {
 		cmgr.TagPeer(p, "kbucket", 5)
 	}
+
 	rt.PeerRemoved = func(p peer.ID) {
 		cmgr.UntagPeer(p, "kbucket")
+		go func(rtRecoveryChan chan *rtRecoveryReq) {
```

Comment:
If we don't, we dealock on the Routing Table lock because the rt.RemovePeer is waiting for this callback to return and  rt.Size() also needs the same lock.